api/envoy/config/core/v3/protocol.proto

```diff
@@ -568,6 +568,12 @@ message Http2ProtocolOptions {
   // Send HTTP/2 PING frames to verify that the connection is still healthy. If the remote peer
   // does not respond within the configured timeout, the connection will be aborted.
   KeepaliveSettings connection_keepalive = 15;
+
+  // [#not-implemented-hide:] Hiding so that the field can be removed after oghttp2 is rolled out.
```

Comment:
Not sure how comfortable we should be with use not-implemented-hide as a general purpose "mutable API" facility. It's technically true that we can do what is described here without it being a breaking API change, but also not sure why this is actually needed if we have a runtime flag and RTDS. CC @envoyproxy/api-shepherds

Reply:
I guess this allows having a per-cluster or per-HCM enablement knob.
An alternative could be an RTDS defined set (of cluster names or listener names) that will use the oghttp2 codec.

Reply:
See #25456 for prior art. The runtime flag is needed so that oghttp2 can eventually be enabled by default, then the flag deprecated and the nghttp2 dependency removed entirely. And Adi is right, this config is needed for more gradual rollout.

Reply:
This is similar to the option added in https://github.com/envoyproxy/envoy/pull/25456. Some Envoy deployments are not able to use a per-instance on/off switch.

Reply:
OK, seems reasonable. That said, I feel this is really a Google special that we're doing here - nobody else can rely on this and you need to coordinate across servers and clients to use this effectively. I understand why we're doing it and because Google is sponsoring the work it seems reasonable, but it's not a pattern I think we should generally be encouraging.

Reply:
I agree it is far from perfect, but the alternative of having a permanent field is worse IMHO.

If we see that this is needed in multiple instances, I can envision a new a mechanism that adds a enable-runtime-feature-set (generalizing the runtime-feature-flags for sets of strings).